### ISSUE TITLE
Bug: RDesign's sample function does not actually perform any sampling

### ISSUE THREAD

**Issue by chaitjo** [A4Bio/RDesign#5]
https://github.com/A4Bio/RDesign/blob/fa410a916f942a0dab7fb4b14d4e2342bcc88fff/model/rdesign_model.py#L76

The RDesign model class seems to implement a method called sample(), which is used during inference/testing. However, the sample() method does not actually seem to use any sampling and directly returns raw logits as output. 

This is an issue because:
1. This means that any design produced by RDesign is deterministic (one input --> one output sequence) and not diverse. 
2. All recovery metric results reported in the paper are not correct (at least based on how recovery is computed in protein design).

Could you clarify whether my understanding is correct, and if so, how one should implementing sampling in RDesign?

For reference, here is how other inverse folding tools implement sampling:
- ProteinMPNN: https://github.com/dauparas/ProteinMPNN/blob/main/protein_mpnn_utils.py#L1180
- ESM-IF: https://github.com/facebookresearch/esm/blob/main/esm/inverse_folding/gvp_transformer.py#L136
- gRNAde: https://github.com/chaitjo/geometric-rna-design/blob/main/src/models.py#L366

**Comment by forrest-yijie-zhang** [comment]
Thanks for looking deep into our code. It is deterministic, as determined by our model's features, thus the sample function did not actually "sample" a sequence. Our recovery calculatation matches the way it was stated in the paper.
